### PR TITLE
Fix Preintegration Covariance of CombinedImuFactor

### PR DESCRIPTION
**Pull request by varunagrawal**
~~This PR serves as a discussion thread for debugging the Preintegration Covariance of the CombinedImuFactor.~~

This PR provides a set of unit tests for the `CombinedImuFactor` and a new `CombinedScenarioRunner` class which is the `ScenarioRunner` equivalent for the `CombinedImuFactor`.

### CODE REVIEW

File: gtsam/navigation/CombinedImuFactor.cpp [hunk 1]
```diff
@@ -134,18 +134,19 @@ void PreintegratedCombinedMeasurements::integrateMeasurement(
   Eigen::Matrix<double, 15, 15> G_measCov_Gt;
   G_measCov_Gt.setZero(15, 15);
 
-  Matrix3 aCov_updated = (aCov + p().biasAccOmegaInt.block<3, 3>(0, 0));
+  Matrix3 aCov_updated = aCov + p().biasAccOmegaInt.block<3, 3>(0, 0);
+  Matrix3 wCov_updated = wCov + p().biasAccOmegaInt.block<3, 3>(3, 3);
```
Comment (varunagrawal) [inline gtsam/navigation/CombinedImuFactor.cpp:138]: We need to understand what is happening here.

File: gtsam/navigation/CombinedImuFactor.cpp [hunk 2]
```diff
@@ -155,7 +156,9 @@ void PreintegratedCombinedMeasurements::integrateMeasurement(
   Matrix3 temp = vel_H_biasAcc * p().biasAccOmegaInt.block<3, 3>(3, 0)
       * theta_H_biasOmega.transpose();
   D_v_R(&G_measCov_Gt) = temp;
+  D_v_t(&G_measCov_Gt) = vel_H_biasAcc * (aCov / dt) * pos_H_biasAcc.transpose();
   D_R_v(&G_measCov_Gt) = temp.transpose();
+  D_t_v(&G_measCov_Gt) = pos_H_biasAcc * (aCov / dt) * vel_H_biasAcc.transpose();
```
Comment (varunagrawal) [inline gtsam/navigation/CombinedImuFactor.cpp:161]: We missed these off-diagonal elements. I guess one test would be to do the full matrix multiplication and see the results, and then update accordingly.

File: gtsam/navigation/tests/testCombinedImuFactor.cpp [hunk 3]
```diff
@@ -44,207 +44,272 @@ static boost::shared_ptr<PreintegratedCombinedMeasurements::Params> Params() {
...
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01;
+
+//   // regression
+//   EXPECT(assert_equal(expected, actual.preintMeasCov()));
+// }
```
Comment (varunagrawal) [inline gtsam/navigation/tests/testCombinedImuFactor.cpp:270]: I commented out all the previous tests to make debugging this faster and easier.

**Comment by varunagrawal** [comment]
Pushed the current draft of the updated `ImuFactor.pdf` here.

**Comment by leCaTo** [comment]
Hii, I'm very interested in testing out the Preintegration+CombinedImuFactor implementation. Are the fixes and updated docs ready to be merged? I can provide some feedback afterwards, thanks!

**Comment by sia79** [comment]
Ecc

**Comment by varunagrawal** [comment]
We're waiting on #882 after which this should be ready to go.

**Comment by varunagrawal** [comment]
@lucacarlone I had forgotten I had added a bunch of unit tests here for exactly what you were asking for in #882.

**Comment by lucacarlone** [comment]
> @lucacarlone I had forgotten I had added a bunch of unit tests here for exactly what you were asking for in #882.

ok thanks!

**Comment by varunagrawal** [comment]
@lucacarlone whenever you get a chance, I'd love to get a review from you on this (especially since you are most familiar with the changes). I'd like to land this and #874 as soon as possible. :slightly_smiling_face:

**Comment by varunagrawal** [comment]
#874 is super easy to review. Only the `CombindImuFactor.cpp` file needs to be reviewed.

**Comment by lucacarlone** [comment]
> #874 is super easy to review. Only the `CombindImuFactor.cpp` file needs to be reviewed.

@varunagrawal  I should be able to get to this by Wednesday (and hopefully before then)

**Review by lucacarlone — APPROVED** [review]
@varunagrawal this looks good to me but I did not re-review the math.

**Pull request merged**